packages/grub/grub.spec

```diff
@@ -74,12 +78,42 @@ Summary: Tools for the bootloader with support for Linux and more
 %description tools
 %{summary}.
 
+%global gnulib_fixes gnulib-fixes-0e9febb5e
+
 %prep
-%autosetup -n grub-%{version} -p1
+rpm2cpio %{S:0} | cpio -iu grub-%{version}.tar.xz \
+  bootstrap bootstrap.conf \
+  gitignore %{gnulib_fixes}.tar.gz \
+  "*.patch"
+
+# Mimic prep from upstream spec to prepare for patching.
+tar -xof grub-%{version}.tar.xz; rm grub-%{version}.tar.xz
+%setup -TDn grub-%{version}
+mv ../bootstrap{,.conf} .
+mv ../gitignore .gitignore
+tar -xof ../%{gnulib_fixes}.tar.gz; rm ../%{gnulib_fixes}.tar.gz
+mv %{gnulib_fixes} gnulib
+pushd gnulib
+patch -p1 < ../../gnulib-amzn2-cflags.patch
+rm ../../gnulib-amzn2-cflags.patch
+popd
 cp unicode/COPYING COPYING.unicode
+rm -f configure
+
+# Apply upstream and local patches.
```

Comment:
Is it required to use git to apply the patches?

Reply:
It's not required, but it's convenient and leaves the tree in a "clean" state with all patches applied as commits.

That's helpful for rebasing patches since you can run just the prep phase (`rpmbuild -bp <spec>`) and then work with the Git repo.